lib/fluent/plugin/out_http.rb

```diff
@@ -1,52 +1,104 @@
+
+  # Simple rate limiting: ignore any records within `rate_limit_msec`
+  # since the last one.
+  config_param :rate_limit_msec, :integer, :default => 0
 
   # Raise errors that were rescued during HTTP requests?
   config_param :raise_on_error, :bool, :default => true
 
-  # Set Net::HTTP.verify_mode to `OpenSSL::SSL::VERIFY_NONE`-  
-  config_param :ssl_no_verify, :bool, :default => false
+  # ca file to use for https request
+  config_param :cacert_file, :string, :default => ''
+
+  # custom headers
+  config_param :custom_headers, :hash, :default => nil
+
+  # 'none' | 'basic' | 'jwt' | 'bearer'
+  config_param :authentication, :enum, list: [:none, :basic, :jwt, :bearer],  :default => :none
+  config_param :username, :string, :default => ''
+  config_param :password, :string, :default => '', :secret => true
+  config_param :token, :string, :default => ''
+  # Switch non-buffered/buffered plugin
+  config_param :buffered, :bool, :default => false
+
+  config_section :buffer do
+    config_set_default :@type, DEFAULT_BUFFER_TYPE
+    config_set_default :chunk_keys, ['tag']
+  end
+
+  config_section :format do
+    config_set_default :@type, DEFAULT_FORMATTER
+  end
 
   def configure(conf)
+    compat_parameters_convert(conf, :buffer, :formatter)
     super
 
+    @ssl_verify_mode = if @ssl_no_verify
+                         OpenSSL::SSL::VERIFY_NONE
+                       else
+                         OpenSSL::SSL::VERIFY_PEER
+                       end
+
+    @ca_file = @cacert_file
+    @last_request_time = nil
+    raise Fluent::ConfigError, "'tag' in chunk_keys is required." if !@chunk_key_tag && @buffered
+
     http_methods = [:get, :put, :post, :delete]
     @http_method = if http_methods.include? @http_method.intern
-                    @http_method.intern
-                  else
-                    :post
-                  end
+                     @http_method.intern
+                   else
+                     :post
+                   end
     puts @endpoint_url
     @uri = URI.parse(@endpoint_url)
-    ssl_verify = @ssl_no_verify
+    # ssl_verify = @ssl_no_verify
     url = @uri.scheme  + "://" + @uri.host + ":" + @uri.port.to_s
     @adapter = Faraday.new(url: url, ssl: {verify:false} ) do |f|
-                 f.request :retry, max:                 5,
-                                   interval:            1,
-                                   interval_randomness: 0.5,
-                                   backoff_factor:      2,
-                                   methods:             @http_method,
-                                   exceptions:          %w(Errno::ETIMEDOUT
-                                                           Faraday::TimeoutError
-                                                           Faraday::Error::TimeoutError
-                                                           Net::ReadTimeout).freeze
+      f.request :retry, max:                 5,
```

Comment:
I think you have an alignment issue here. You can read more about my arguments against aligning the code this way all the way from 2015 [by following this thread on Slack](https://gimbal.slack.com/archives/C0310JK3K/p1435875939000136).